### DISCUSSION TITLE
Kobo Clara 2E

### DISCUSSION THREAD

**Discussion by psa-avat** [XCSoar/XCSoar#1089] [General]
Hi
 I'm presently adding the Kobo Clara HD 2E to XCSoar.
 With few modifications, it's almost working.
 I still have some issues with the WIFI and I'am also planning to add BT.
 I saw that some drivers are profided separtely from the main code of XCSoar. 
 I don't know yet if I'll need to work on them but if needed, is anyone can give me some clue to build them ?

Patrick

**Comment by ubx** [comment]
Hello Patrick
As a hint for BT: for the Libra 2, (which seems to be similar to the Clara HD 2E)  I wrote some [scripts](https://github.com/ubx/XCSoarData/tree/master/kobo/scripts) for the Tools Menu.

Andreas

**Comment by psa-avat** [comment]
Hello Andreas,

  Thanks, I will check them out.
  I also wrote some scripts on my side for different use cases.
  I finished the code for the CLARA 2e and will try to push it soon.

Patrick

Le mer. 28 déc. 2022 à 10:17, Andreas Lüthi ***@***.***> a
écrit :

**Comment by psa-avat** [comment]
Hello Andreas

  I tested your scripts but the Bluetooth wont work.
  I think the hardware is different from others kobo.
  It looks like they use a single nxp chip for both wifi and BT.
  I had to modify the way to activate the wifi.

   For BT the ttymcx1 is the good serial port

dmesg | grep uart
imx-uart 2020000.serial: could not get sleep state
imx-uart 2024000.serial: assigned as bt port host  *this is /dev/ttymxc1*

  Every time I run the hciattach I got a time out error

  Any idea ?


Patrick



///
case KoboModel::CLARA_2E:
    InsMod("/drivers/mx6sll-ntx/wifi/sdio_wifi_pwr.ko");
    InsMod("/drivers/mx6sll-ntx/wifi/mlan.ko");
    InsMod("/drivers/mx6sll-ntx/wifi/moal.ko",
"fw_name=nxp/sdiouart8987_combo_v0.bin","cfg80211_wext=0xf",
"cal_data_cfg=none");
    break ;




 .

**Reply by ubx** [reply]
Hi Patrick

>   Every time I run the hciattach I got a time out error

There is no hciattach in my script. Do you mean the line `/sbin/rtk_hciattach -n -s 115200 ttymxc1 rtk_h5 &` in `connect.sh` ?

Andreas

**Comment by psa-avat** [comment]
Got the same issue with rtk
It's dedicated to Realtek chipset

Le sam. 14 janv. 2023, 13:15, Andreas Lüthi ***@***.***> a
écrit :

**Reply by ubx** [reply]
You could start Nickel, switch on BT and show which process is running. On the Libra 2 `ps` shows 
` 1754 root      0:00 /sbin/rtk_hciattach -n -s 115200 ttymxc1 rtk_h5`

Here is a little description of how to [prepare the Kobo for testing and debugging](https://gist.github.com/ubx/a1b507f597d1e401de2df25e4d8f3224). Maybe this helps.

**Comment by psa-avat** [comment]
Hi

 I saw, you changed your connect script.
 I was also working on it and got the same result for hciattach.
 I made some modifications on xcsoar to make it working on clara 2E.
 I think, I'll make a pull request soon, but we can share in advance if
you want.


Patrick


Le dim. 15 janv. 2023 à 12:51, Andreas Lüthi ***@***.***> a
écrit :
